File: 229-majority-element-ii/majority-element-ii.cpp
class Solution {
public:
    vector<int> majorityElement(vector<int>& nums) {
        unordered_map<int,int> mp;
        vector<int> ans;
        for(int i : nums){
            mp[i]++;
        }
        int n = nums.size()/3;
        for(int i = 0;i < nums.size();i++){
            if(mp[nums[i]] > n){
                ans.push_back(nums[i]);
                mp[nums[i]] = 0;
            }
        }
        return ans;
    }
};

File: 229-majority-element-ii/majority-element-ii.java
class Solution {
    public List<Integer> majorityElement(int[] nums) {
        HashMap<Integer,Integer> mp = new HashMap<>();
        ArrayList<Integer> ans = new ArrayList<>();
        for(int i : nums){
            mp.put(i,mp.getOrDefault(i,0)+1);
        }
        int n = nums.length/3;
        for(int i = 0;i < nums.length;i++){
            if(mp.get(nums[i]) > n){
                ans.add(nums[i]);
                mp.put(nums[i],0);
            }
        }
        return ans;
    }
}